### PR TITLE
[@types/update-notifier] Use Boxen's options type directly

### PR DESCRIPTION
**Pull request by astorije**
This adds support for, for example: `margin: { top: 0, left: 3, bottom: 1 },`

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
  - ⚠️ `yarn tsc types/update-notifier/index.d.ts` was successful after I added `@types/configstore` to `package.json` locally.

~~⚠️ Halp, how can I add `boxen` as a dependency somewhere? There is no `package.json` so I'm not sure how to compile/test. Thanks! 🙏~~  Done!

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/sindresorhus/boxen/blob/master/index.d.ts>
- If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/update-notifier/index.d.ts [hunk 1]
```diff
@@ -7,6 +7,7 @@
 //                 Piotr Błażejewicz <https://github.com/peterblazejewicz>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
+import boxen = require('boxen');
```
Comment (peterblazejewicz) [inline types/update-notifier/index.d.ts:10]: this requires `package.json` to be added with the PR.
Adding dependency is OK, as the `update-notifier` depends on `boxen`

Comment (astorije) [reply types/update-notifier/index.d.ts:10]: Awesome, thanks, I'll look into it! Apologies for the dumb mistakes, DefinitelyTyped noob here!

Out of curiosity, do you know why `configstore` below didn't need a `package.json`?

Comment (astorije) [reply types/update-notifier/index.d.ts:10]: Oh I think that'd be it:

> A package.json may be included to specify dependencies that are not other `@types` packages.

From https://github.com/DefinitelyTyped/DefinitelyTyped#i-notice-some-packages-having-a-packagejson-here

`configstore` is a `@types` package: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/configstore/index.d.ts

Comment (astorije) [reply types/update-notifier/index.d.ts:10]: Done at https://github.com/microsoft/DefinitelyTyped-tools/pull/74

**Comment by astorije** [comment]
For reference, CI currently fails with:

> Error: In package.json: Dependency boxen not in the allowed dependencies list.
> If you are depending on another `@types` package, do *not* add it to a `package.json`. Path mapping should make the import work.
> For namespaced dependencies you then have to add a `paths` mapping from `@namespace/library` to `namespace__library` in `tsconfig.json`.
If this is an external library that provides typings,  please make a pull request to microsoft/DefinitelyTyped-tools adding it to `packages/definitions-parser/allowedPackageJsonDependencies.txt`.

That should disappear once https://github.com/microsoft/DefinitelyTyped-tools/pull/74 gets merged.

File: types/update-notifier/index.d.ts [hunk 2]
```diff
@@ -61,7 +54,8 @@ declare namespace UpdateNotifier {
         /** Include the -g argument in the default message's npm i recommendation */
         isGlobal?: boolean;
         /** Options object that will be passed to `boxen` */
```
Comment (peterblazejewicz) [inline types/update-notifier/index.d.ts:56]:
```suggestion
```

File: types/update-notifier/index.d.ts [hunk 3]
```diff
@@ -61,7 +54,8 @@ declare namespace UpdateNotifier {
...
-        boxenOptions?: BoxenOptions;
+        /** See https://github.com/sindresorhus/boxen/blob/master/index.d.ts */
```
Comment (peterblazejewicz) [inline types/update-notifier/index.d.ts:57]:
```suggestion
        /**
         * Options object that will be passed to `boxen`
         * See https://github.com/sindresorhus/boxen/blob/master/index.d.ts
         */
```

Comment (astorije) [reply types/update-notifier/index.d.ts:57]: Oops, duh yes, sorry 🤦

**Comment by peterblazejewicz** [comment]
@astorije 
Now with everything in place, this works:
```ts
    notifier.notify({
        message: 'Update available: ' + notifier.update.latest,
        defer: false,
        isGlobal: true,
        boxenOptions: {
            padding: 1,
            margin: 1,
            borderColor: 'yellow',
            backgroundColor: 'magenta',
            borderStyle: {
                topLeft: '+',
                topRight: '+',
                bottomLeft: '+',
                bottomRight: '+',
                horizontal: '-',
                vertical: '|'
            },
        },
    });
```
while this one (also taken from boxen sample), does not:
```ts
notifier.notify({
        message: 'Update available: ' + notifier.update.latest,
        defer: false,
        isGlobal: true,
        boxenOptions: {
            padding: 1,
            margin: 1,
            borderColor: '#eebbaa',
            borderStyle: 'double',
        },
    });
```
the last one requires terse syntax:
```ts
notifier.notify({
        message: 'Update available: ' + notifier.update.latest,
        defer: false,
        isGlobal: true,
        boxenOptions: {
            padding: 1,
            margin: 1,
            borderColor: '#eebbaa',
            borderStyle: BorderStyle.Double
        },
    });
```
Is that OK?

**Comment by astorije** [comment]
Excellent find, @peterblazejewicz, thanks! I think that's ok because that's how you'd use `boxen` directly using TypeScript, `borderStyle: 'double'` wouldn't work... I think? (https://github.com/sindresorhus/boxen/blob/master/index.d.ts#L162 says otherwise, but I don't see how that could be).
What do you think?

**Comment by peterblazejewicz** [comment]
For me that's ok, that's a nuance of using TS with enums. If `borderStyle` could be rewritten to string unions, it would work out-of-the-box. 

Let's wait a day or two for someone else opinion. For me the change is OK

**Comment by astorije** [comment]
Sounds good! We need that other PR to get merged first anyway. Cheers!

**Comment by astorije** [comment]
That other PR got merged, I'm assuming it needs a release now.

**Comment by astorije** [comment]
I'm not sure how to fix the failing tests here, any suggestions?

**Comment by peterblazejewicz** [comment]
> I'm not sure how to fix the failing tests here, any suggestions?

Try this locally:
```diff
❯ git diff
diff --git a/types/update-notifier/index.d.ts b/types/update-notifier/index.d.ts
index 628a5960a5..94a536adff 100644
--- a/types/update-notifier/index.d.ts
+++ b/types/update-notifier/index.d.ts
@@ -6,7 +6,7 @@
 //                 Michael Grinich <https://github.com/grinich>
 //                 Piotr B<C5><82>a<C5><BC>ejewicz <https://github.com/peterblazejewicz>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-
+// TypeScript Version: 3.2
 import boxen = require('boxen');
 import ConfigStore = require('configstore');

diff --git a/types/update-notifier/tsconfig.json b/types/update-notifier/tsconfig.json
index 7ce9885785..53de995251 100644
--- a/types/update-notifier/tsconfig.json
+++ b/types/update-notifier/tsconfig.json
@@ -3,7 +3,8 @@
         "module": "commonjs",
         "lib": [
             "es6",
-            "dom"
+            "dom",
+            "esnext"
         ],
         "noImplicitAny": true,
         "noImplicitThis": true,
```
it seems it's because of the 3rd party dependency used by Boxen (see CI test complaining about 3rd aprty lib and esnext new features).

**Review by peterblazejewicz — COMMENTED** [review]
@vvakame @nchen63 @bitjson @grinich
thoughts?

**Comment by astorije** [comment]
Thanks for your help, @peterblazejewicz!!

**Review by andrewbranch — APPROVED** [review]
Seems fine to me 👍

**Comment by astorije** [comment]
Ready to merge

**Pull request merged**